File: src_qt/rules/ClearMemoryNode.cpp
#include <QDebug>

#include "ClearMemoryNode.h"

#include "../Brain.h"

namespace CnotiMind
{

	ClearMemoryNode::ClearMemoryNode( const MemoryType& memory, Brain* brain, QObject* parent ) :
		RuleNode(brain, parent),
		_memory( memory )
	{
	}


	void ClearMemoryNode::exec()
	{
		switch( _memory )
		{
		case LongTermMemory:
			_brain->clearMemory();
			break;
		case WorkingMemory:
			_brain->clearWorkingMemory();
			break;
		}
	}

	void ClearMemoryNode::exec( QHash<QString, QString>& variables )
	{
		Q_UNUSED( variables )

		exec();
	}

	QString ClearMemoryNode::info( int depth ) const
	{
		return space( depth ) + "ClearMemory (" + (_memory == LongTermMemory ? "LTM" : "WM" ) + ")";
	}

	ClearMemoryNode *ClearMemoryNode::fromXML(const QString &qName, const QXmlAttributes &atts,
											  Brain *brain, QObject *parent)
	{
		if( qName.compare( "ClearMemory", Qt::CaseInsensitive ) == 0 )
		{
			MemoryType memory = translateMemoryType( atts.value( "memory" ) );

			return new ClearMemoryNode( memory , brain, parent );
		}
		return NULL;
	}
}

File: src_ios/CnotiMind/rules/ClearMemoryNode.m
//
//  ClearMemoryNode.m
//  SingleEmotion
//
//  Created by Gonçalo Rodrigues on 8/31/11.
//  Copyright 2011 Universidade de Coimbra. All rights reserved.
//

#import "ClearMemoryNode.h"
#import "Brain.h"

@implementation ClearMemoryNode

- (id)init
{
    self = [super init];
    if (self) {
        // Initialization code here.
    }
    
    return self;
}


- (id) initWithMemoryAndBrainAndParent: (enum MemoryType)aMemory 
                                 brain:(Brain*)aBrain 
                                parent:(id)aParent
{
    if (self = [super initWithBrainAndParent:aBrain parent:aParent]) {
        _memory = aMemory;
    }
    
    return self;
}


- (void) exec
{
    switch (_memory) {
        case LongTermMemory:
            [_brain clearMemory];
            break;
        case WorkingMemory:
            [_brain clearWorkingMemory];
            break;
            
        default:
            break;
    }

}


- (void) exec:(NSMutableDictionary*)aVariables{
    [self exec];
}


- (NSString*) info:(int)aDepth{
    NSString* info = [NSString stringWithFormat:@""];
    NSString* space = [self space:aDepth];
    
    info = [info stringByAppendingFormat:@"%@ ClearMemory (%@)", space, _memory == LongTermMemory ? @"LTM" : @"WM"];
    info = [info stringByAppendingFormat:@"%@", [super info:aDepth]];
    
    return info;
}


- (void)dealloc
{
    [super dealloc];
}


@end
